### ISSUE TITLE
release-tag pre-compiled binaries for armv7

### ISSUE THREAD

**Issue by jdubz93** [google/yggdrasil-decision-forests#10]
Could we get some pre-compiled binaries for armv7. I'm currently trying to compile on a rpi 3b and stuck on:

    Compiling org_tensorflow/tensorflow/core/framework/node_def_util.cc; 12299s local
    Compiling org_tensorflow/tensorflow/core/util/batch_util.cc; 12280s local
    [Sched] Compiling org_tensorflow/tensorflow/core/util/tensor_slice_set.cc; 12299s
    [Sched] Compiling org_tensorflow/tensorflow/core/util/matmul_autotune.cc; 12280s

Note I used the `config=use_tensorflow_io` flag which I think installs the tensorflow io libraries. However makes the compiling much longer. I had to do it this way because of an error I received using default settings. Also had to take out the 
``` bash
# Instruction set optimizations
build:linux_avx2 --copt=-mavx2
```
because it would error out with -maxvx2 is a unknown command.

Also unrelated question: 
is there anyway we could get hypertuner support for tensorflow decision forest? 
https://www.tensorflow.org/tutorials/keras/keras_tuner#instantiate_the_tuner_and_perform_hypertuning

Like on sklearn you have gridsearchcv, keras NN's have hypertuner. Would be nice to have this optimization for decision forests.

Best Regards,

**Comment by janpfeifer** [comment]
Just to clarify: the request on raspeberry pi (armv7) is for inference only.

On tuner: Keras Tuner should work, I used it a few weeks ago.  Let us know if not. Link:

https://www.tensorflow.org/tutorials/keras/keras_tuner

Now, having said that, doing a tutorial page on tuning TF-DF models is also in our TODO list.

**Comment by jdubz93** [comment]
Awesome! Thank you Jan! :D

Below is the latest logs from trying to cross compile using an armv7 compiler. Like you mentioned the library requires a ton of dependencies and we are not sure if they dependencies will work. If you guys have time could you create a stand alone example of just inference. That would really be helpful in understanding which libraries are requires / not needed.
 
`I am trying to compile on gcc-9 armv7 compiler and getting this error:
arm-linux-gnueabihf-gcc-9.3.0: error: unrecognized command line option ‘-msse3’

ERROR: /home/jw/.cache/bazel/bazel_jw/ec70e549ffe7593eeda7bb3167603cae/external/org_tensorflow/tensorflow/core/platform/profile_utils/BUILD:42:11: Compiling tensorflow/core/platform/profile_utils/cpu_utils.cc [for host] failed: (Exit 1): arm-linux-gnueabihf-gcc-9.3.0 failed: error executing command
(cd /home/jw/.cache/bazel/bazel_jw/ec70e549ffe7593eeda7bb3167603cae/execroot/main &&
exec env -
LD_LIBRARY_PATH=/mnt/pi-toolchain/sysroot/usr/lib
PATH=’/home/jw/.local/bin:/home/jw/.nvm/versions/node/v14.17.4/bin:/home/jw/miniconda3/bin:/home/jw/.local/bin:/home/jw/miniconda3/bin:/home/jw/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0:/mnt/c/Program Files/Intel/WiFi/bin:/mnt/c/Program Files/Common Files/Intel/WirelessCommon:/mnt/c/WINDOWS/System32/OpenSSH:/mnt/c/Program Files/dotnet:/mnt/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/mnt/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/mnt/c/Program Files (x86)/Intel/Intel(R) Management Engine Components/DAL:/mnt/c/Program Files/Intel/Intel(R) Management Engine Components/DAL:/mnt/c/Users/jw/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/jw/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/jw/.dotnet/tools:/snap/bin:/mnt/pi-toolchain/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf/bin’
PWD=/proc/self/cwd
/home/jw/cross_gcc_9/cross-pi-gcc-9.3.0-2/bin/arm-linux-gnueabihf-gcc-9.3.0 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 ‘-D_FORTIFY_SOURCE=1’ -DNDEBUG -ffunction-sections -fdata-sections ‘-std=c++0x’ -MD -MF bazel-out/host/bin/external/org_tensorflow/tensorflow/core/platform/profile_utils/objs/profile_utils_cpu_utils/cpu_utils.d '-frandom-seed=bazel-out/host/bin/external/org_tensorflow/tensorflow/core/platform/profile_utils/objs/profile_utils_cpu_utils/cpu_utils.o’ -DEIGEN_MPL2_ONLY ‘-DEIGEN_MAX_ALIGN_BYTES=64’ -iquote external/org_tensorflow -iquote bazel-out/host/bin/external/org_tensorflow -iquote external/eigen_archive -iquote bazel-out/host/bin/external/eigen_archive -iquote external/com_google_absl -iquote bazel-out/host/bin/external/com_google_absl -iquote external/nsync -iquote bazel-out/host/bin/external/nsync -isystem external/org_tensorflow/third_party/eigen3/mkl_include -isystem bazel-out/host/bin/external/org_tensorflow/third_party/eigen3/mkl_include -isystem external/eigen_archive -isystem bazel-out/host/bin/external/eigen_archive -isystem external/nsync/public -isystem bazel-out/host/bin/external/nsync/public -g0 -w -g0 ‘-std=c++14’ -DEIGEN_AVOID_STL_ARRAY -Iexternal/gemmlowp -Wno-sign-compare ‘-ftemplate-depth=900’ -fno-exceptions -DINTEL_MKL -msse3 -DTENSORFLOW_MONOLITHIC_BUILD -pthread -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE=“redacted”’ '-D__TIMESTAMP=“redacted”’ ‘-D__TIME__=“redacted”’ -c external/org_tensorflow/tensorflow/core/platform/profile_utils/cpu_utils.cc -o bazel-out/host/bin/external/org_tensorflow/tensorflow/core/platform/profile_utils/_objs/profile_utils_cpu_utils/cpu_utils.o)
Execution platform: @local_execution_config_platform//:platform
arm-linux-gnueabihf-gcc-9.3.0: error: unrecognized command line option ‘-msse3’
Target //:beginner_cc failed to build`

**Comment by achoum** [comment]
Yggdrasil can now be compiled and run (both the training and inference) on Arm7.

[This documentation](https://github.com/google/yggdrasil-decision-forests/blob/main/documentation/installation.md#compilation-on-and-for-raspberry-pi) shows how to compile and run it on a Raspberry Pi 4 Model B Rev 1.4.

I hope this help :)
